app/tasks.py

```diff
@@ -59,7 +59,27 @@ def ocr_task(self, pdf_bytes, strategy_name, pdf_hash, ocr_cache, prompt, model)
             num_chunk += 1
             extracted_text += chunk['response']
 
-    self.update_state(state='DONE', meta={'progress': 100 , 'status': 'Processing done!', 'start_time': start_time, 'elapsed_time': time.time() - start_time})  # Example progress update
+    # Optional call to generate tags and summary
+    if prompt and model:
+        tags_summary = generate_tags_summary(prompt, model)
+        extracted_text += "\n\nTags and Summary:\n" + tags_summary
```

Comment:
When generate_metadata option defined it should be not returned within the general output but stored in the different json filename according to #10, used for file name strategies and stored within other field in the celerey result

We probably need an additional endpoint to get just the metadata stored for specific celery task